### ISSUE TITLE
RuntimeError: Input type (torch.cuda.ByteTensor) and weight type (torch.cuda.FloatTensor) should be the same

### ISSUE THREAD

**Issue by francesco-taioli** [open-mmlab/mmtracking#390]
**Describe the bug**
RuntimeError: Input type (torch.cuda.ByteTensor) and weight type (torch.cuda.FloatTensor) should be the same

**Reproduction**
```
python demo/demo_mot_vis.py configs/mot/bytetrack/bytetrack_yolox_x_crowdhuman_mot17-private-half.py \
     --input myvideo.mp4 \
     --output res.mp4
```

2. Did you make any modifications on the code or config? Did you understand what you have modified? **No modifications**

**Environment**

```
sys.platform: linux
Python: 3.7.11 (default, Jul 27 2021, 14:32:16) [GCC 7.5.0]
CUDA available: True
GPU 0,1: GeForce RTX 2080 Ti
CUDA_HOME: /usr/local/cuda-11.1
NVCC: Build cuda_11.1.TC455_06.29190527_0
GCC: gcc (Ubuntu 5.5.0-12ubuntu1) 5.5.0 20171010
PyTorch: 1.8.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.9.0
OpenCV: 4.5.5
MMCV: 1.4.2
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 11.1
MMTracking: 0.9.0+56c133e
```

I had to upgrade numpy to ```numpy==1.20.3```, see `[this](https://stackoverflow.com/a/66066058/5723524)`

**Error traceback**
```
_bootstrap.py:219: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88, got 80
...
Warning: The model doesn't have classes
[                                                  ] 0/39, elapsed: 0s, ETA:Traceback (most recent call last):
  File "demo/demo_mot_vis.py", line 100, in <module>
    main()
  File "demo/demo_mot_vis.py", line 75, in main
    result = inference_mot(model, img, frame_id=i)
  File "mypath/mmtrack/apis/inference.py", line 94, in inference_mot
    result = model(return_loss=False, rescale=True, **data)
  File "my_virtual_env/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "my_virtual_env/lib/python3.7/site-packages/mmcv/runner/fp16_utils.py", line 98, in new_func
    return old_func(*args, **kwargs)
  File "mypath/mmtrack/models/mot/base.py", line 136, in forward
    return self.forward_test(img, img_metas, **kwargs)
  File "mypath/mmtrack/models/mot/base.py", line 113, in forward_test
    return self.simple_test(imgs[0], img_metas[0], **kwargs)
  File "mypath/mmtrack/models/mot/byte_track.py", line 66, in simple_test
    img, img_metas, rescale=rescale)
  File "my_virtual_env/lib/python3.7/site-packages/mmdet/models/detectors/single_stage.py", line 101, in simple_test
    feat = self.extract_feat(img)
  File "my_virtual_env/lib/python3.7/site-packages/mmdet/models/detectors/single_stage.py", line 43, in extract_feat
    x = self.backbone(img)
  File "my_virtual_env/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "my_virtual_env/lib/python3.7/site-packages/mmdet/models/backbones/csp_darknet.py", line 281, in forward
    x = layer(x)
  File "my_virtual_env/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "my_virtual_env/lib/python3.7/site-packages/mmdet/models/backbones/csp_darknet.py", line 64, in forward
    return self.conv(x)
  File "my_virtual_env/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "my_virtual_env/lib/python3.7/site-packages/mmcv/cnn/bricks/conv_module.py", line 201, in forward
    x = self.conv(x)
  File "my_virtual_env/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "my_virtual_env/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "my_virtual_env/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 396, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: Input type (torch.cuda.ByteTensor) and weight type (torch.cuda.FloatTensor) should be the same
my_virtual_env/lib/python3.7/tempfile.py:798: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/tmp/tmpgorhfmd1'>
  _warnings.warn(warn_message, ResourceWarning)
```

**Other**
This error only appears with [ByteTrack](https://mmtracking.readthedocs.io/en/latest/model_zoo.html#bytetrack-arxiv-2021), the others [multiple object tracking](https://mmtracking.readthedocs.io/en/latest/model_zoo.html#baselines-of-multiple-object-tracking) run as expected.

**Comment by jandremarais** [comment]
Just started with open-mmlab software today so probably not the best to respond but I did notice that here the input is still of type uint8 when fed to the model for a forward pass and I suspect it should be of type float. I think this conversion is usually taken care of by the Normalization transform which YOLOX does not use. As a hack I added a normalization transform to the test pipeline but using zero mean and unit variance. This gives me a result but I can't say whether it's optimal.
